tests/e2e/etcd_config_test.go

```diff
@@ -358,4 +359,23 @@ func TestBootstrapDefragFlag(t *testing.T) {
 	if err = proc.Stop(); err != nil {
 		t.Fatal(err)
 	}
+
+	// wait for the process to exit, otherwise test will have leaked goroutine
+	proc.Close()
+}
+
+func TestSnapshotCatchupEntriesFlag(t *testing.T) {
+	e2e.SkipInShortMode(t)
+
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	proc, err := e2e.SpawnCmd([]string{e2e.BinPath.Etcd, "--experimental-snapshot-catchup-entries", "1000"}, nil)
+	require.NoError(t, err)
+	require.NoError(t, e2e.WaitReadyExpectProc(ctx, proc, []string{"\"snapshot-catchup-entries\":1000"}))
+	require.NoError(t, e2e.WaitReadyExpectProc(ctx, proc, []string{"serving client traffic"}))
+	require.NoError(t, proc.Stop())
```

Comment:
(just side note) currently, the proc.Stop just sends SIGTERM signal to process. basically, it won't return error.
The `proc.Close` might return error if there is somewrong in server.

Reply:
Yeah, sending SIGTERM to the process will lead to the exit code to be `-1`

https://github.com/etcd-io/etcd/blob/5b163aa50752c08705a0e8d8b6a6b36beb054b16/pkg/expect/expect.go#L280-L282

```
ExitCode returns the exit code of the exited process, or -1 if the process hasn't exited or was terminated by a signal. 
```

`proc.Stop` can be potentially improved to include cleanly wait until process exited.